### PR TITLE
Meal info component

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description

what this pull request is doing

Modify the meal card modal to use the MUI react component library for rendering the modal

what will be done in later PRs and not included here

Implement a consistent style across the website through context

## Milestones
<the milestones or stories/features that this works towards>
Frontend converting website into react components

## Resources
<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>
https://mui.com/joy-ui/react-modal/
https://mui.com/joy-ui/react-typography/
https://mui.com/joy-ui/react-box/
https://mui.com/joy-ui/react-list/

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>
Images included in google doc

https://docs.google.com/document/d/1u4QuX_0hLM_7-EKXGbdEtpxEErOixgLAIE5qFVQna7s/edit?usp=sharing

### CODE REVIEW

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -37,55 +39,28 @@ model IngredientsOnHand {
...
-  endDate   DateTime
-  toBuy     Ingredient[]
-  userId    Int          @unique
-  user      User         @relation(fields: [userId], references: [id])
+  id           Int      @id @default(autoincrement()) // store the id number from spoonacular api, otherwise id is auto incremented (part of the users database)
+  apiId        Int      @unique
+  recipeTitle  String
+  previewImage String // if image not provided, generic image will be presented
+  servings     Int
+  ingredients  Json
```
Comment (radhikadhomse) [inline backend/prisma/schema.prisma:57]: question - do we want to store this as a json or have list of ingredients and transform to json when rendering?
 
is this more of an efficiency optimization so we dont have to convert each time? 
do you ever have to update the recipe?

File: frontend/src/components/TitledListView.tsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ const TitledListView = <T,>({
           ))}
         </div>
       )}
-      <div className={className}>{list.map(renderItem)}</div>
```
Comment (radhikadhomse) [inline frontend/src/components/TitledListView.tsx:27]: low priority feedback: 
in a separate PR, can you change this and make sure you're using more clear naming 
i would avoid words like "list" because in many languages it's a keyword so it gets confusing

File: frontend/src/components/MealInfoModal.tsx [hunk 3]
```diff
@@ -17,44 +24,63 @@ const MealInfoModal: React.FC<GPMealModalProps> = ({
...
+              {recipeInfo?.dairyFree && <Typography sx={listItemStyle}>Dairy Free</Typography>}
+              {recipeInfo?.glutenFree && <Typography sx={listItemStyle}>Gluten Free</Typography>}
+              {recipeInfo?.vegetarian && <Typography sx={listItemStyle}>Vegetarian</Typography>}
+              {recipeInfo?.vegan && <Typography sx={listItemStyle}>Vegan</Typography>}
+            </Box>
+          </Box>
+        </Box>
+        <Typography level="h3">Ingredients</Typography>
+        <List marker="circle">
+          {recipeInfo?.ingredients.map((ingredient) => {
```
Comment (radhikadhomse) [inline frontend/src/components/MealInfoModal.tsx:59]: so technically, if recipeInfo?.ingredients is undefined, you will still have an issue when map is called

Comment (radhikadhomse) [reply frontend/src/components/MealInfoModal.tsx:59]: {(recipeInfo?.ingredients ?? []).map((ingredient)  maybe is safer

**Review by radhikadhomse — APPROVED** [review]
very small comments, otherwise LGTM
